### PR TITLE
Pdf

### PR DESCRIPTION
**Pull request by VanessaPC**
## Urgency
- [ ] Blocker 
- [x] High
- [ ] Medium

## Requirements
- Test all commands and procedures.
- Add [redirects](https://github.com/mesosphere/dcos-docs-site/wiki/Redirects).
- Change all affected versions (e.g. 1.7, 1.8, 1.9, 1.10, 1.11).
- See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/wiki/Contributing).

### CODE REVIEW

File: scripts/mdtopdf.sh [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        NEW_FILE="${d}/${FILE_NAME}"
+        if [ -f $NEW_FILE ]
+        then
+        # Create temporary file with all md content to send to pandoc // this avoids very long urls & long strings (Pandoc has a limit)
+          TEMP_FILE=$(mktemp)
+          TEMP_FILES="${TEMP_FILES} ${TEMP_FILE}"
+
+          cat ${NEW_FILE} >> ${TEMP_FILE}
+        fi
+      done
```
Comment (aayvazyan) [inline scripts/mdtopdf.sh]: can we please change the way how we find files from for loop to somewhat  above ?
 while IFS= read -r -d '' SOURCE_FILE
  do
     ...
     ...
  done <  <(find "${INPUT_FOLDER}" -type f -name "*.md" -print0)

File: scripts/mdtopdf.sh [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  echo "======="
+  echo "Starting pdf build $(date)"
+  cat "${PARALLEL_TEMPFILE}" | parallel --halt-on-error 2 --progress --eta --workdir "${PWD}" --jobs "${PARALLEL_JOBS:-4}"
+  echo "Finished build $(date)"
+  echo $TEMP_FILES
+  # rm -f "${TEMP_FILES}"
+}
+
+clean "${OUTPUT_FOLDER}"
+main "${INPUT_FOLDER}" "${OUTPUT_FOLDER}"
```
Comment (aayvazyan) [inline scripts/mdtopdf.sh]: Makes sense to run a shellcheck and fix the major issues?
https://github.com/koalaman/shellcheck

File: scripts/mdtopdf.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+     #Fix for all local links to not mislead Pandoc into grabbing other files when there is nothing to look for
+     # issue reported in Pandoc : https://github.com/jgm/pandoc/issues/3619
+     sed -i -r 's,]:,]\\:,g' "${TEMP_FILE}"
+     sed -i -r 's,\/#,,g' "${TEMP_FILE}"
+
+    # Set name for last folder
+    if [ -z "${PDF_FILE_NAME}" ]
+    then
+      "${PDF_FILE_NAME}"="MesosphereDCOS"
```
Comment (fatz) [inline scripts/mdtopdf.sh]: i think this should be 

`PDF_FILE_NAME="MesosphereDCOS"` correct?

**Comment by fatz** [comment]
Added a few code comments I was struggling with locally.

Ok some thoughts.

First of all I really like the result. Its a pretty good PDF having TOC and links. Also page numbers and Chapter titles. Its way better than what we had before.

Downside is that the PDF build takes ages (at least on my machine) However that should not stup us. Performance improvements could be done later.

This is IMO not a show stopper as we didn't had this before BUT
- The chapter should be Adjusted. Currently the header line are always of the same value h1, h2 etc.
  - I think as a first improvement we should maybe try to use the yaml "title" as a latex chapter. 
  - A second improvement could be folder ordering by numbers. Like in `https://github.com/progit/progit/tree/master/en`
- Another improvment would be a kind of title page for the version documentation.

So lets say in general LGTM! We should migrate this to dev and take a look into the result. So @pavisandhu could decide if it is ready to be shipped.

File: scripts/mdtopdf.sh (hunk 3 above)
Comment (VanessaPC) [reply scripts/mdtopdf.sh]: Correct, I'll do that change today.

File: Makefile [hunk 4]
```diff
@@ -34,17 +34,11 @@ build-development: build-api
 #
 
 build-pdf-production:
-	./scripts/pdf.sh ./build ./build-pdf
-
-build-pdf-concat-production:
-	./scripts/pdf-concat.sh ./pages ./build-pdf
+	./scripts/mdtopdf.sh ./pages ./build-pdf
 
 build-pdf-development: build-api docker-development-up-pdf
```
[the remark this answers is not in the document]
Comment (VanessaPC) [reply Makefile]: These changes have been made in the last commit.

File: scripts/mdtopdf.sh (hunk 3 above)
Comment (VanessaPC) [reply scripts/mdtopdf.sh]: This is also fixed as per the second to last commit.

**Pull request merged**